pkg/dsl/definition/template.go

```diff
@@ -119,6 +121,14 @@ func (td *traitDef) Complete(ctx process.Context) error {
 			return errors.WithMessagef(err, "traitDef %s build", td.name)
 		}
 
+		processing := inst.Lookup("processing")
```

Comment:
I wonder why does this have to be part of the rendering process?

Reply:
The logic of the rendering is currently too simple, just is data passing as converter. In fact, the trait will exec complicated data processing instead of directly convert, including http, exec, and so on.

Reply:
This is by design, the rendering part is not supposed to know the meaning of what each trait/workload is doing at all. We need to decouple the rendering logic from the trait/workload logic. 

With that said, there can be some exceptions, and an HTTP task might be one of them.